app/questionnaire/questionnaire_schema.py

```diff
@@ -1248,6 +1262,29 @@ def get_item_anchor(self, section_id: str, list_name: str) -> str | None:
                 if item["for_list"] == list_name and item.get("item_anchor_answer_id"):
                     return f"#{str(item['item_anchor_answer_id'])}"
 
+    def _populate_placeholder_section_dependencies(self) -> None:
+        for section in self.get_sections():
```

Comment:
Can we not do one for loop and something like we did before:
```python
    def _populate_answer_dependencies(self) -> None:
        for block in self.get_blocks():
```
and whenever you need section (id) just call
```python
self.get_section_for_block_id(block_id)
```

Reply:
Updated using `self.get_blocks() `🖖